emperor/support_files/js/emperor.js

```diff
@@ -873,15 +873,15 @@ function displayTaxaLabels(taxaLevel){
 
         $('#taxalabels').css('display','block');
         $("#taxaLevel" ).html(taxonomy[taxaLevel]);
-        for(var key in g_taxaPositions){
+        _.each(g_taxaPositions, function(taxaPos, key){
             // get the coordinate of this taxa sphere
             var coords = toScreenXY(g_plotTaxa[key].position,g_sceneCamera,$('#main-plot'));
 
-            // labels are identified by the key they have in g_taxaPositions
-            labelshtml += "<label id=\""+key+"_taxalabel\" class=\"unselectable labels\" style=\"position:absolute; left:"+parseInt(coords['x'])+"px; top:"+parseInt(coords['y'])+"px;\">";
-            labelshtml += truncateLevel(g_taxaPositions[key]['lineage'], taxaLevel);
+            // labels are identified by the key they have in taxaPos
```

Comment:
I think changing this for append (as in the other _.each) will be nicer. Note that you will need to remove the document... line after the each.